tests/unit/via/views/api/youtube_test.py

```diff
@@ -1,29 +1,55 @@
+import logging
+from unittest.mock import sentinel
+
 import pytest
-from h_matchers import Any
-from pyramid.httpexceptions import HTTPNotFound
 
 from via.views.api import youtube
 
 
 class TestTranscript:
-    def test_it(self, pyramid_request):
+    def test_it(self, pyramid_request, youtube_service):
         response = youtube.transcript(pyramid_request)
 
+        youtube_service.get_transcript.assert_called_once_with(sentinel.video_id)
         assert response == {
             "data": {
                 "type": "transcripts",
-                "id": "1",
-                "attributes": {"segments": Any.list()},
+                "id": sentinel.video_id,
+                "attributes": {"segments": youtube_service.get_transcript.return_value},
             }
         }
 
-    def test_it_errors_if_the_video_id_is_unknown(self, pyramid_request):
-        pyramid_request.matchdict["video_id"] = "unknown"
+    def test_it_handles_errors_from_youtube(
+        self, caplog, pyramid_request, youtube_service, report_exception_to_sentry
+    ):
+        exception = youtube_service.get_transcript.side_effect = RuntimeError(
+            "Test error"
+        )
 
-        with pytest.raises(HTTPNotFound):
-            youtube.transcript(pyramid_request)
+        response = youtube.transcript(pyramid_request)
+
+        report_exception_to_sentry.assert_called_once_with(exception)
+        assert caplog.record_tuples == [
+            ("via.views.api.youtube", logging.ERROR, str(exception))
+        ]
+        assert pyramid_request.response.status_int == 500
+        assert response == {
+            "errors": [
+                {
+                    "status": 500,
+                    "code": "failed_to_get_transcript",
+                    "title": "Failed to get transcript from YouTube",
+                    "detail": str(exception),
```

Comment:
```suggestion
                    "detail": str(exception).strip(),
```

The exceptions from the library have leading space for some reason?

Reply:
:+1: https://github.com/hypothesis/via/pull/1010/commits/92a5b8fc16e8d410f4fdf83d1bf57c984f8998eb